engine/class_modules/priest/sc_priest_shadow.cpp

```diff
@@ -205,8 +185,8 @@ struct mind_blast_t final : public priest_spell_t
 
     if ( priest().buffs.dark_thoughts->up() )
       priest().buffs.dark_thoughts->decrement();
-    else
-      priest_spell_t::update_ready( cd_duration );
+
+    priest_spell_t::update_ready( cd_duration );
```

Comment:
Is this right? If you decrement Dark Thoughts, that's gonna remove one charge and then update_ready will start the cooldown, removing another charge.